operations/mimir/ingester.libsonnet

```
@@ -56,6 +56,20 @@
 
   ingester_env_map:: {
     JAEGER_REPORTER_MAX_QUEUE_SIZE: '1000',
+
+    local requests = $.util.parseCPU($.ingester_container.resources.requests.cpu),
+    local requestsWithHeadroom = std.ceil(
+      // double the requests, but with headroom of at least 3 and at most 6 CPU
+      // too little headroom can lead to throttling
+      // too much headroom can lead to setting the limit above the available CPU cores
+      requests + std.max(3, std.min(6, requests)),
```

Wouldn't this _potentially_ set it to more than what is available for the host/container? E.g. if a VM has only 4 cores, and there is only one container running on the host (kinda line in AWS fargate), we set maxprocs to 7?

---

very good point, Peter already thought about that 🙂

*  #8201

---

Oh, I didn't know that 👍  

Maybe it's worth it to link to clamping PR from the comment, to make sure this doesn't trigger anyone later

```suggestion
      // double the requests, but with headroom of at least 3 and at most 6 CPU
      // too little headroom can lead to throttling
      // too much headroom can lead to setting the limit above the available CPU cores
      // if the value exceeds the number of available CPU cores on the host, mimir caps the maxprocs on application start (ref grafana/mimir#8201)
      requests + std.max(3, std.min(6, requests)),
```